plugins/tiledb/io/TileDBReader.cpp

```diff
@@ -255,6 +264,42 @@ void TileDBReader::ready(PointTableRef)
         m_query->set_subarray(subarray);
     }
 
+    // read spatial reference
+    NL::json meta = nullptr;
+
+#if TILEDB_VERSION_MAJOR >= 1 && TILEDB_VERSION_MINOR >= 7
+    tiledb_datatype_t v_type = TILEDB_UINT8; // CPLSerializeXMLTree returns char*
+    const void* v_r;
+    uint32_t v_num;
+    m_array->get_metadata("_pdal", &v_type, &v_num, &v_r);
+    if (v_r != NULL)
+        meta = NL::json::parse(static_cast<const char*>(v_r));
+#endif
+
+    if (meta == nullptr)
+    {
+        tiledb::VFS vfs(*m_ctx, m_ctx->config());
+        tiledb::VFS::filebuf fbuf(vfs);
+        std::string metaFName = m_arrayName + pathSeparator + "pdal.json";
```

Comment:
Maybe I'm missing something, but if what you're trying to do is read the contents of `pdal.json` into a string, you can just use `pdal::FileUtils::readFileIntoString(filename)`

Reply:
@abellgithub I use VFS for the cases where this file in on S3, this PR moves away from that to use internal array metadata, this code is there to keep compatibility with older library versions.